src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -37,15 +38,23 @@ final class ArgumentResolver implements ArgumentResolverInterface
     public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFactory = null, iterable $argumentValueResolvers = [])
     {
         $this->argumentMetadataFactory = $argumentMetadataFactory ?? new ArgumentMetadataFactory();
-        $this->argumentValueResolvers = $argumentValueResolvers ?: self::getDefaultArgumentValueResolvers();
+        $this->argumentValueResolvers = $argumentValueResolvers ?: iterator_to_array(self::getIndexedDefaultArgumentValueResolvers());
```

Comment:
using a generator here looks overkill to transform a simple flat array

Reply:
It’s the best way I know to index an array based on its value :sweat_smile: